tests/phpunit/tests/test-preload-paths.php

```diff
@@ -89,4 +89,43 @@ public function test_preload_path_for_translatable_media() {
 			'Media path should be filtered by language when option is activated.'
 		);
 	}
+
+	/**
+	 * @ticket #1861 {@see https://github.com/polylang/polylang-pro/issues/1861}
+	 */
+	public function test_rest_routes_for_custom_taxonomies() {
+		register_post_type( 'book' );
+
+		register_taxonomy(
+			'genre',
+			'book',
+			array(
+				'show_in_rest' => true,
+			)
+		);
+
+		$this->pll_admin->model->cache->clean( 'post_types' );
+		$this->pll_admin->model->cache->clean( 'taxonomies' );
+
+		add_filter(
+			'pll_get_post_types',
+			function ( $post_types ) {
+				return array_merge( $post_types, array( 'book' => 'book' ) );
+			}
+		);
+
+		add_filter(
+			'pll_get_taxonomies',
+			function ( $taxonomies ) {
+				return array_merge( $taxonomies, array( 'genre' => 'genre' ) );
+			}
+		);
```

Comment:
```suggestion
		register_taxonomy_for_object_type( 'trtax', 'trcpt' );
```
The test case already defines translated post types and a translated taxonomy with the same parameter so we should not need to add new ones.
Here I propose to redefine to add the previously defined taxonomy to the previously defined cpt instead of creating new ones. However, it my even be better to directly define the taxonomy for this cpt in the test case if doesn't break other tests or require only small adaptation.